infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/StateTransfer.java

```diff
@@ -221,7 +221,6 @@ private static void transferChunk(Layout layout,
                 throw new IllegalStateException("Missing address");
             }
             LogData ld = (LogData) dataMap.get(address);
-            ld.setPayloadCodecType(Codec.Type.valueOf(runtime.getParameters().getCodecType()));
```

Comment:
I actually did this intentionally; assuming that the client and the server bits have the same codecs. The reason is, to load the new server/client version with an old data log (i.e. uncompressed), and the data can be compressed by bouncing the servers one-by-one to trigger state transfer and compress the data without having to rewrite it via checkpoint. Maybe that's not a deal breaker, but its just something for you to consider.